README.md

```diff
@@ -1,31 +1,201 @@
+  - Does not define parameters or type of parameters (this is for a later iteration) - for now, you can add this information to the description
+- Added tag events with name, description
+  - Lists all events the custom elements fires via `this.dispatchEvent()`
+- Added tag slots with name, description
+  - Lists all [slots](https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element) in the shadow dom like `<slot name="header"></slot>`
+  - The default slot is represented via an empty name `<slot></slot>`
+- Added tag cssProperties with name, description, type
+  - Lists all [CSS custom properties](https://drafts.csswg.org/css-variables/#defining-variables) that are used within the style of a web component like `--my-css-property-name`
+  - Type are aligned with [Houdini's CSS Properties and Values API](https://drafts.css-houdini.org/css-properties-values-api)
+- Added tag cssShadowParts with name, description
+  - Lists all [CSS Shadow parts](https://www.w3.org/TR/css-shadow-parts-1/)
+- Added tag demos with name, url
+  - List all demos you have for your web component
+
+#### Specification
+
+In the example above you see all of the most common cases however if you need more details take a look at the
+
+- [typescript schema](./schema.ts)
+- [json schema](./schema.json)
 
-## Editor Support
+You can use these schemas to validate against your own `custom-elements.json`.
+There is also a [full example](./example.json) which you can test the validation on.
+There are various online tools which allow you to validate directly in the browser
 
-Developers using custom elements should be able to get full-featured IDE support including auto-competion, hover-documentation, unknown symbol warnings, etc. These features should be available in HTML files, and in various template syntaxes via template-specific tools.
+- [jsonschemalint](https://jsonschemalint.com/#/version/draft-07/markup/json)
+- [json-schema-validator](https://json-schema-validator.herokuapp.com/syntax.jsp)
 
-## Documentation
+Or you can use one of the available validators for javascript
+
+- [ajv](https://github.com/epoberezkin/ajv)
+- [djv](https://github.com/korzio/djv)
+- [...](https://json-schema.org/implementations.html#validator-javascript)
+
+Or many other languages.
+
+## Use Cases
+
+### Editor Support
+
+Developers using custom elements should be able to get full-featured IDE support including auto-completion, hover-documentation, unknown symbol warnings, etc. These features should be available in HTML files, and in various template syntaxes via template-specific tools.
+
+### Documentation
 
 Documentation viewers should be able to display all the relevant information about a custom element, such as its tag name, attributes, properties, definition module, CSS variables and parts, etc.
 
-## Linting
+### Documentation Playground
+
+With full information about the public api of a web component tools can provide UIs to trigger/execute these apis.
```

Comment:
"public api of a web component tools can" - I think you need a comma to break this up: "public api of a web component, tools can..."